### PR TITLE
fix(k8sprocessor): owner meta deletion grace period

### PR DESCRIPTION
**Pull request by c-kruse**
Adds a deferred deletion queue to the owner metadata cache, allowing time for telemetry data to be processed with correct metadata after the owning resource(s) have been deleted.

Closes https://github.com/SumoLogic/sumologic-otel-collector/issues/1242

### CODE REVIEW

File: pkg/processor/k8sprocessor/kube/owner.go [hunk 1]
```diff
@@ -469,3 +489,37 @@ func (op *OwnerCache) GetOwners(pod *Pod) []*ObjectOwner {
...
+			op.deleteMu.Lock()
+			for i, d := range op.deleteQueue {
+				if d.ts.Add(gracePeriod).After(now) {
+					break
+				}
+				cutoff = i + 1
+			}
+			toDelete := op.deleteQueue[:cutoff]
+			op.deleteQueue = op.deleteQueue[cutoff:]
+			op.deleteMu.Unlock()
```
Comment (echlebek) [inline pkg/processor/k8sprocessor/kube/owner.go:511]: I feel this is worth its own method, and then the Unlock can be deferred

**Review by echlebek — COMMENTED** [review]
There seems to be a large amount of commented out code in one of the test files that this PR touches. I would suggest deleting it if possible.

**Comment by c-kruse** [comment]
@swiatekm-sumo you cool with yanking the commented out benchmarks? They had drifted a bit even before this CL. Saw that you had been "using" them in earlier PR/issue discussion.

**Review by swiatekm-sumo — COMMENTED** [review]
The overall approach looks good to me. I'd like:
1. Comments for the new functions
2. A test that actually measures the deletion delay
3. Maybe a manual E2E test in Kubernetes to make sure this is ok? I'm not sure this is necessary, but it's not too difficult to do using the Helm Chart integration tests. Let me know if you want to try it, but if not, I'm ok merging this change as-is.

**Comment by c-kruse** [comment]
@swiatekm-sumo Updated godocs and tests with assertions on honoring the grace period.

Have dug into the E2E tests and may yet manually validate the change.

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**